dumpling/export/dump.go

```diff
@@ -994,38 +1009,33 @@ func selectTiDBTableRegion(tctx *tcontext.Context, conn *sql.Conn, meta TableMet
 			pkVals = append(pkVals, []string{pkVal})
 		}
 		return nil
+	}, func() {
+		pkFields = pkFields[:0]
+		pkVals = pkVals[:0]
 	}, tableRegionSQL, dbName, tableName)
 
 	return pkFields, pkVals, errors.Trace(err)
 }
 
-func selectTiDBPartitionRegion(tctx *tcontext.Context, conn *sql.Conn, dbName, tableName, partition string) (pkVals [][]string, err error) {
-	var (
-		rows      *sql.Rows
-		startKeys []string
-	)
+func selectTiDBPartitionRegion(tctx *tcontext.Context, conn *BaseConn, dbName, tableName, partition string) (pkVals [][]string, err error) {
+	var startKeys [][]string
 	const (
 		partitionRegionSQL = "SHOW TABLE `%s`.`%s` PARTITION(`%s`) REGIONS"
 		regionRowKey       = "r_"
 	)
 	logger := tctx.L().With(zap.String("database", dbName), zap.String("table", tableName), zap.String("partition", partition))
-	rows, err = conn.QueryContext(tctx, fmt.Sprintf(partitionRegionSQL, escapeString(dbName), escapeString(tableName), escapeString(partition)))
-	if err != nil {
-		err = errors.Trace(err)
-		return
-	}
-	startKeys, err = GetSpecifiedColumnValueAndClose(rows, "START_KEY")
+	startKeys, err = conn.QuerySQLWithColumns(tctx, []string{"START_KEY"}, fmt.Sprintf(partitionRegionSQL, escapeString(dbName), escapeString(tableName), escapeString(partition)))
 	if err != nil {
 		return
 	}
 	for rowID, startKey := range startKeys {
-		if rowID == 0 {
+		if rowID == 0 || len(startKey) != 1 {
```

Comment:
why `len(startKey) != 1` is added

Reply:
Because we use `GetSpecifiedColumnValuesAndClose` to replace `GetSpecifiedColumnValueAndClose`. Actually `len(startKey)` is impossible to be 1.